src/Compiler/Common.idr

```diff
@@ -342,30 +342,38 @@ exists f
              | Left err => pure False
          closeFile ok
          pure True
-
--- Parse a calling convention into a backend/target for the call, and
--- a comma separated list of any other location data.
+-- Select the most preferred target from an ordered list of choices and
+-- parse the calling convention into a backend/target for the call, and
+-- a comma separated list of any other location data. For example
+-- the chez backend would supply ["scheme,chez", "scheme", "C"]. For a function with
+-- more than one string, a string with "scheme" would be preferred over one
+-- with "C" and "scheme,chez" would be preferred to both.
 -- e.g. "scheme:display" - call the scheme function 'display'
 --      "C:puts,libc,stdio.h" - call the C function 'puts' which is in
 --      the library libc and the header stdio.h
--- Returns Nothing if the string is empty (which a backend can interpret
--- however it likes)
+-- Returns Nothing if there is no match.
 export
-parseCC : String -> Maybe (String, List String)
-parseCC "" = Nothing
-parseCC str
-    = case span (/= ':') str of
-           (target, "") => Just (trim target, [])
-           (target, opts) => Just (trim target,
-                                   map trim (getOpts
-                                       (assert_total (strTail opts))))
+parseCC : List String -> List String -> Maybe (String, List String)
```

Comment:
Can we document the meanings of the two arguments that happen to have the same type?